packages/stores/src/ui-config/convert-to-stake.ts

```diff
@@ -0,0 +1,264 @@
+        // only share pools
+        const queryPool = this.osmosisQueries.queryPools.getPool(poolId);
+        if (Boolean(queryPool?.sharePool)) return true;
+        return false;
+      });
+
+    const account = this.osmosisAccount;
+    const stakeCurrency = this.chainGetter.getChain(
+      this.osmosisChainId
+    ).stakeCurrency;
+
+    if (!account) return [];
+
+    const conversions: SuggestedConvertToStakeAssets[] = [];
+
+    // calculate conversions for user's share pools
+    ownedSharePoolIds.forEach((sharePoolId) => {
+      const { sharePoolDetail, superfluidPoolDetail, poolBonding } =
+        this.derivedDataStore.getForPool(sharePoolId);
+
+      // only include if contains stake currency
+      if (
+        !sharePoolDetail.querySharePool?.hasPoolAsset(
+          stakeCurrency.coinMinimalDenom
+        )
+      )
+        return;
+
+      // only include pools with > 1¢ value
+      // TODO: temporarily only look at bonded value, since we can't serialize this case in the message
+      if (
+        !(
+          sharePoolDetail.userStats?.bondedValue?.toDec().isPositive() ||
+          sharePoolDetail.userUnlockingAssets.length > 0
+        )
+      )
+        return;
+
+      const totalValue =
+        sharePoolDetail.userStats?.totalShareValue ??
+        new PricePretty(this.fiatCurrency, 0);
+      const userPoolAssets = sharePoolDetail.userPoolAssets.map(
+        ({ asset }) => asset
+      );
+
+      const currentApr =
+        poolBonding.highestBondDuration?.aggregateApr ??
+        superfluidPoolDetail.superfluidApr.add(sharePoolDetail.swapFeeApr);
+
+      // only include if better opportunity
+      if (currentApr.toDec().lte(this.stakeApr.toDec().quo(new Dec(100)))) {
+        conversions.push({
+          poolId: sharePoolId,
+          totalValue,
+          userPoolAssets,
+          currentApr,
+        });
+      }
+    });
+
+    return conversions;
+  }
+
+  @computed
+  get isConvertToStakeFeatureRelevantToUser() {
+    return this.suggestedConvertibleAssetsPerPool.length > 0;
+  }
+
+  @computed
+  get hasValidatorPreferences() {
+    // TODO: returning false since we can't serialize this case in the message
+    // this case prevents the message from containing an empty validator to defer to valset of delegations
+    return false;
+
+    // return this.osmosisQueries.queryUsersValidatorPreferences.get(
+    //   this.accountAddress
+    // ).hasValidatorPreferences;
+  }
+
+  /** Returns true if the user has at least some delegation in the SDK staking module. */
+  @computed
+  get hasDelegation() {
+    // TODO: returning false since we can't serialize this case in the message
+    // this case prevents the message from containing an empty validator to defer to valset of delegations
+    return false;
+
+    // return (
+    //   this.cosmosQueries.queryDelegations.getQueryBech32Address(
+    //     this.accountAddress
+    //   ).delegations.length > 0
+    // );
+  }
```

Comment:
Is it safe to merge this to prod without these functional boolean flags?

Reply:
Yeah, as these flags are used to ensure the user selects a validator every time.

Then since a validator is selected, we won't have to use a null value for the validator address in the message, therefore avoiding the nullable value signature error